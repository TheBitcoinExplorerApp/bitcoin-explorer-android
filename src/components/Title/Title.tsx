/* eslint-disable import/no-extraneous-dependencies */
import { StyleSheet, View } from "react-native";
import { useState } from "react";
import LogoIcons from "../../assets/iconsLogo.svg";
import Modal from "../Modal/Modal";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    gap: 26,
  },
  title: {
    color: "#DF7800",
    fontSize: 20,
    fontWeight: "600",
  },
  brand: {
    color: "#DF7800",
    fontSize: 28,
    fontWeight: "bold",
  },
});

export default function Title() {
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <View style={styles.container}>
      <LogoIcons
        height={32}
        onPress={() => {
          setModalVisible(true);
        }}
      />

      <Modal
        modalType="Information"
        isVisible={modalVisible}
        handleModalClose={() => {
          setModalVisible(false);
        }}
      />
    </View>
  );
}
