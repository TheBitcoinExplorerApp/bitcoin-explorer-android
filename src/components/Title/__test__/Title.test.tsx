// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import Title from "../Title";

describe("Title", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Title />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
