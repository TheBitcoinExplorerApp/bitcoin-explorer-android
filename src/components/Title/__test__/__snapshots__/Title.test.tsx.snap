// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 26,
      "justifyContent": "center",
    }
  }
>
  <svg
    height={32}
  />
</View>
`;
